android-kit-base/src/main/java/com/mparticle/kits/KitIntegrationFactory.java

```diff
@@ -15,64 +16,82 @@
 public class KitIntegrationFactory {
 
     final Map<Integer, Class> supportedKits = new HashMap<>();
+    final static Map<Integer, LocalKit> localKits = new HashMap<>();
 
-    public KitIntegrationFactory() {
+    public KitIntegrationFactory(MParticleOptions options) {
+        mergeIntegrations(options);
         loadIntegrations();
     }
 
+    public static int generateRandomKey() {
```

Comment:
Why generate random IDs? See the solution on iOS, we just defined a number (`1000000000`) that is the start of the sideloaded kits range, so that it's trivial to check if a kit is sideloaded or not just by checking it's ID without needing to reference a known list of IDs.